chainer/iterators/multiprocess_iterator.py

```diff
@@ -0,0 +1,158 @@
+from __future__ import division
+import multiprocessing
+
+import numpy
+import six
+
+from chainer.dataset import iterator
+
+
+class MultiprocessIterator(iterator.Iterator):
+
+    """Dataset iterator that loads examples in parallel.
+
+    This is an implementation of :class:`~chainer.dataset.Iterator` that loads
+    examples with worker processes. It uses the standard :mod:`multiprocessing`
+    module to parallelize the loading. In order to access the dataset by worker
```

Comment:
"The dataset is sent to worker process in the standard way that uses pickle." is simpler.